database/drivers/postgres/parse.go

```diff
@@ -202,14 +271,78 @@ func queryPrimaryKeys(log *log.Logger, db *sql.DB, schemas []string) ([]*databas
 	return ret, nil
 }
 
+type indexResult struct {
+	SchemaName string
+	TableName  string
+	IndexName  string
+	Columns    []string
+}
+
+func queryIndexes(log *log.Logger, db *sql.DB, schemaNames []string) ([]indexResult, error) {
+	const q = `
+	SELECT
+		n.nspname as schema,
+		i.indrelid::regclass as table,
+		c.relname as name,
+		array_to_string(ARRAY(
+			SELECT pg_get_indexdef(i.indexrelid, k + 1, true)
+			FROM generate_subscripts(i.indkey, 1) as k
+			ORDER BY k
+		), ',') as column_names
+	FROM pg_index as i
+	JOIN pg_class as c
+		ON c.oid = i.indexrelid
+	JOIN pg_namespace as n
+		ON n.oid = c.relnamespace
+	WHERE n.nspname IN (%s)`
+
+	spots := make([]string, len(schemaNames))
+	vals := make([]interface{}, len(schemaNames))
+	for i := range schemaNames {
+		spots[i] = fmt.Sprintf("$%v", i+1)
+		vals[i] = schemaNames[i]
+	}
+
+	query := fmt.Sprintf(q, strings.Join(spots, ", "))
+	rows, err := db.Query(query, vals...)
+	defer rows.Close()
+	if err != nil {
+		return nil, errors.WithMessage(err, "error querying indexes")
+	}
+
+	var results []indexResult
+	for rows.Next() {
+		var r indexResult
+		var cs string
+		if err := rows.Scan(&r.SchemaName, &r.TableName, &r.IndexName, &cs); err != nil {
+			return nil, errors.WithMessage(err, "error scanning index")
+		}
+		r.Columns = strings.Split(cs, ",") // array converted to string in query
+
+		// postgres prepends schema onto table name
+		r.TableName = r.TableName[strings.LastIndex(r.TableName, ".")+1:]
```

Comment:
if we have schema name scanned from the DB, could we use the length of that to slice the table name, rather than searching for the period?  They're probably equally valid, but using the name as given seems just a tiny bit less likely to hit edge cases.